opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -352,8 +352,7 @@ def __init__(
         instrumentation_info: "InstrumentationInfo",
     ):
         self.instrumentation_info = instrumentation_info
-        self.processor = Processor(source.stateful)
-        self.resource = source.resource
+        self.processor = Processor(source.stateful, source.resource)
```

Comment:
Take a look at [this comment](https://github.com/open-telemetry/opentelemetry-python/pull/1153#discussion_r497193591). I believe we should actually decouple `Tracer` from `TracerProvider`, and equivalently, `Meter` from `MeterProvider` by removing the `source` attribute. We should be passing in configuration through the `Provider` into the constructor of both `Tracer` and `Meter`. `Resource` would be one of those configurations. Not sure if  you want to do this as part of this PR however.

Reply:
It looks like issue https://github.com/open-telemetry/opentelemetry-python/issues/1181 already tracks this work, I'd rather do this as a follow up to this PR.